### PR TITLE
convert windmill min timestamp to beam min timestamp

### PR DESCRIPTION
**Pull request by Naireen**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Add a link to the appropriate issue in your description, if applicable. This will automatically link the pull request to the issue.

### CODE REVIEW

**Comment by Naireen** [comment]
R: @scwhittle

File: runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/WindmillTimeUtilsTest.java [hunk 1]
```diff
@@ -56,6 +56,7 @@ public void testWindmillToHarnessTimestamp() {
     assertEquals(new Instant(-17), windmillToHarnessTimestamp(-16987));
     assertEquals(new Instant(-17), windmillToHarnessTimestamp(-17000));
     assertEquals(new Instant(-18), windmillToHarnessTimestamp(-17001));
+    assertEquals(BoundedWindow.TIMESTAMP_MIN_VALUE, windmillToHarnessTimestamp(Long.MIN_VALUE + 1));
```
Comment (scwhittle) [inline runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/WindmillTimeUtilsTest.java:59]: Did you verify this test fails without the change to WindmillTimeUtils?
Looking again I'm wondering if the divideAndRoundDown would have already ended up with this value.

File: runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillTimeUtils.java [hunk 2]
```diff
@@ -44,6 +44,10 @@ public class WindmillTimeUtils {
   public static Instant windmillToHarnessTimestamp(long timestampUs) {
     // Windmill should never send us an unknown timestamp.
     Preconditions.checkArgument(timestampUs != Long.MIN_VALUE);
+    // Set windmill min timestamp to beam min timestamp
+    if (timestampUs == Long.MIN_VALUE + 1) {
+      return BoundedWindow.TIMESTAMP_MIN_VALUE;
+    }
```
Comment (scwhittle) [inline runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillTimeUtils.java:50]: actually would other values (ie Long.MIN_VALUE + 2) also be problematic (though that timestamp would be less likely to be observed)

maybe this should calculate result as before but then bounds check it with TIMESTAMP.MIN_VALUE similar to how it bound checks TIMESTAMP_MAX_VALUE

**Comment by Naireen** [comment]
Run Java PreCommit

**Comment by scwhittle** [comment]
Run Java PreCommit

**Comment by aaltay** [comment]
Run Java PreCommit

**Comment by aaltay** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by aaltay** [comment]
Run Java PreCommit

**Comment by aaltay** [comment]
@kennknowles - precommit flaked 5 times in a row. We need to do something about it.

**Comment by kennknowles** [comment]
Yes. We have someone already working on "org.apache.beam.sdk.io.pulsar.PulsarIOTest.testReadFromSimpleTopic: Trying to claim offset 1655305408194 while last attempted was 1655305409570" but I think it just needs to be disabled. I also saw a lot of docker issues. Tests that use a local server in a docker container should probably be moved off of the main job. Because the job uses gradle's `buildDependents` it is hard to make fine-grained adjustments. I'm thinking about it.

**Comment by aaltay** [comment]
> Yes. We have someone already working on "org.apache.beam.sdk.io.pulsar.PulsarIOTest.testReadFromSimpleTopic: Trying to claim offset 1655305408194 while last attempted was 1655305409570" but I think it just needs to be disabled. I also saw a lot of docker issues. Tests that use a local server in a docker container should probably be moved off of the main job. Because the job uses gradle's `buildDependents` it is hard to make fine-grained adjustments. I'm thinking about it.

Thank you.

I think I saw other issues as well. But I did not save the jenkins links. IO was the general root cause.

**Comment by kennknowles** [comment]
Hmm this is green but in https://ci-beam.apache.org/view/PostCommit/job/beam_PreCommit_Java_Cron/5560/ and all subsequent tests (including current PRs) we are seeing `org.apache.beam.runners.dataflow.worker.WindmillTimerInternalsTest.testTimerDataToFromTimer` fail with

```
Expected: <TimerData{timerId=0:-9223372036854775, timerFamilyId=, namespace=Global, timestamp=-290308-12-21T19:59:05.225Z, outputTimestamp=-290308-12-21T19:59:05.224Z, domain=EVENT_TIME, deleted=false}>
     but: was <TimerData{timerId=0:-9223372036854775, timerFamilyId=, namespace=Global, timestamp=-290308-12-21T19:59:05.225Z, outputTimestamp=-290308-12-21T19:59:05.225Z, domain=EVENT_TIME, deleted=false}>
```

and based on timing and code touched this PR seems the most likely to be related. TBH I don't get it since this seems like it is deterministic and would have been red on this PR

**Comment by Abacn** [comment]
"timestamp=-290308-12-21T19:59:05.225Z" is a negative year. It is an overflow. Looks like the conversion treated overflow integers in different way.

**Pull request merged**